src/common/transformations/tests/common_optimizations/fuse_moe_test.cpp

```diff
@@ -0,0 +1,335 @@
+    auto up_bmm = makeOP<opset1::MatMul>({batched_input, fused_up_weights->output(0)},
+                                         {{"transpose_a", false}, {"transpose_b", true}});
+    auto swiglu_mul = makeOP<opset1::Multiply>({gate_swish, up_bmm}, {{"auto_broadcast", "numpy"}});
+    auto down_bmm = makeOP<opset1::MatMul>({swiglu_mul, fused_down_weights->output(0)},
+                                           {{"transpose_a", false}, {"transpose_b", true}});
+
+    auto expert_output_shape =
+        makeOP<opset1::Concat>({num_experts_unsqueeze, batch_dim_unsqueeze, minus_one, hidden_dim_unsqueeze},
+                               {{"axis", 0}});
+    auto expert_outputs = makeOP<opset1::Reshape>({down_bmm, expert_output_shape}, {{"special_zero", false}});
+
+    // Create routing weights via scatter to match 3-GEMM pattern
+    auto topk_values = topk_TopK->output(0);
+    auto sum_reduce = makeOP<opset1::ReduceSum>({topk_values, axis_minus_one}, {{"keep_dims", true}});
+    auto normalized_topk =
+        makeOP<opset1::Divide>({topk_values, sum_reduce}, {{"auto_broadcast", "numpy"}, {"m_pythondiv", true}});
+    auto zeros_scalar = makeConst(ov::element::f32, ov::Shape{}, std::vector<float>{0.0f});
+    auto scatter_shape =
+        std::make_shared<ov::op::v0::Concat>(ov::OutputVector{batch_dim_unsqueeze, num_experts_unsqueeze}, 0);
+    auto zeros_tensor = std::make_shared<ov::op::v3::Broadcast>(zeros_scalar, scatter_shape);
+    auto scatter = std::make_shared<ov::op::v12::ScatterElementsUpdate>(zeros_tensor,
+                                                                        topk_TopK->output(1),
+                                                                        normalized_topk,
+                                                                        axis1_vector);
+    auto router_transpose = std::make_shared<ov::op::v1::Transpose>(scatter, transpose_axes);
+    auto router_shape =
+        std::make_shared<ov::op::v0::Concat>(ov::OutputVector{num_experts_unsqueeze, batch_dim_unsqueeze, minus_one},
+                                             0);
+    auto router_reshape = std::make_shared<ov::op::v1::Reshape>(router_transpose, router_shape, true);
+    auto routing_unsqueeze = std::make_shared<ov::op::v0::Unsqueeze>(router_reshape, axis_minus_one_vector);
+
+    // Apply routing and sum
+    auto weighted_outputs = std::make_shared<ov::op::v1::Multiply>(expert_outputs, routing_unsqueeze);
+    auto final_output = std::make_shared<ov::op::v1::ReduceSum>(weighted_outputs, axis0_vector, false);
+
+    auto final_reshape = std::make_shared<ov::op::v1::Reshape>(final_output, target_shape, false);
+    auto final_add = std::make_shared<ov::op::v1::Add>(residual_input, final_reshape);
+    final_add->set_friendly_name("moe_decomposed");
+
+    return std::make_shared<ov::Model>(final_add,
+                                       ov::ParameterVector{final_hidden_states_, router_logits, hidden_states_2d});
+}
+
```

Comment:
Consider to add edge case tests:
  - Different expert counts (1, 2, large numbers like 64, 128)
  - Different topk values (1, topk==expert_num, edge cases)
  - No decompression case (fp32 weights without Convert)
  - Mixed weights (some experts with decompression, some without)
  - Invalid expert_id ordering (non-sequential IDs)
  - Missing experts (gaps in expert IDs, e.g., 0,1,3,4 - missing 2)
  - Mismatched weight dimensions between experts
  - Multiple MoE layers in the same model
  
Should we consider these test cases or they are not possible?